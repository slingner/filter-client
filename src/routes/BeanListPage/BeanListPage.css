.submit-button {
  padding: 10px 20px 10px 20px;
  margin-top: 8px;
  margin-left: 20px;
  min-width: 170px;
  font-size: 15px;
  background-color: rgb(110, 147, 115);
  color: white;
  font-family: 'Montserrat', sans-serif;
}

.submit-button:hover {
  cursor: pointer;
  background-color: rgb(227, 161, 85);
  color: rgb(90, 90, 90);
}

.beanlist-header {
  text-align: left;
  padding: 10px;
  padding-left: 50px;
  font-size: 20px;
  color: rgb(110, 147, 115);
  font-family: 'Montserrat', sans-serif;
}

.BeanList {
  display: flex;
  justify-content: left;
  flex-direction: right;
}

.checkbox-wrapper {
  display: flex;
  /* justify-content: start; */
  margin-top: 90px;
  min-width: 200px;
  align-items: center;
  flex-direction: column;
  border: 4px solid rgba(185, 185, 185, 0.2);
  border-radius: 10px;
  height: fit-content;
  padding: 15px;
  margin-left: 20px;
}

@media screen and (max-width: 600px) {
  .BeanList {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .beanlist {
    display: flex;
    flex-direction: row;
  }

  .submit-button {
    padding: 10px 30px 10px 30px;
    margin-top: 8px;
    min-width: 150px;
    font-size: 15px;
    background-color: rgb(110, 147, 115);
    color: white;
    font-family: 'Montserrat', sans-serif;
  }

  .checkbox-wrapper {
    margin: 15px auto;
  }
}

/* @media screen 
  and (min-width: 1000px) 
  and (max-height: 1366px) {

    .BeanList {
      display: flex;
      flex-direction: column;
      justify-content: center;
    }
  
    .beanlist {
      display: flex;
      flex-direction: row;
    }

    .checkbox-wrapper {
      display: flex;
      justify-content: space-around;
      align-items: center;
      flex-direction: column;
    }

    .submit-button{
      padding: 10px 30px 10px 30px;
      margin-top: 20px;
      min-width: 200px;
      background-color: rgb(110, 147, 115);
      color: white;
    }
} */
