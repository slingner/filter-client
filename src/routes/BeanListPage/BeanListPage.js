import React, { Component } from 'react';
import BeansListContext from '../../contexts/BeansListContext';
import BeanCard from '../BeanCard/BeanCard';
import './BeanListPage.css';
import CheckBox from '../../components/Checkbox/Checkbox';
import config from '../../config';

export default class BeanListPage extends Component {
  state = {
    flavors: [],
    flavorsSelected: new Map(),
    BeanPage: false,
    error: null,
  };

  static contextType = BeansListContext;

//do I need to call for all flavors here? isn't this happening in Checkbox?
  componentDidMount() {
    this.context.fetchAllBeans();
    fetch(`${config.API_ENDPOINT}/allflavors`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json',
      }
    })
      .then(res => {
        if (!res.ok) {
          return res.json().then(error => Promise.reject(error))
        }
        return res.json()
      })
      .then(flavors => {
        this.setState({
          flavors,
          error: null,
        })
      })
      .catch(error => {
        console.error(error)
        this.setState({ error })
      })
  }

  handleChange = (e) => {
    const item = e.target.id;
    const isChecked = e.target.checked;
    this.setState(prevState => ({ flavorsSelected: prevState.flavorsSelected.set(item, isChecked) }));
  }
//this maps through the Map() to retreive the true/checked values
//once retreived, pass results through fetchbean
  handleClick = () => {
    const map = this.state.flavorsSelected
    let array = Array.from(map.entries())
    array = array.filter(val => {
      return val[1] === true
    }).map(val => {
      return val[0]
    })
    this.context.fetchBeansByFlavorId(array)
  }

  render() {
    const { beans } = this.context
    const { flavors } = this.state

    return (
      <section className='BeanList'>
        <div className='checkbox-wrapper'>
            <CheckBox 
              handleChange={this.handleChange}
              checked={this.state.flavorsSelected.get(flavors.flavor_name)}
              handleClick={this.handleClick}
            />    
            <button className='submit-button' onClick={this.handleClick}>Refresh List</button>
        </div>
        
        <div className='beanlist-wrapper' aria-live='polite'>
          <header className='beanlist-header'>LIST OF BEANS</header>
            <div className='beanlist'>
              {beans.map((bean, idx) => {
                  return (
                  <BeanCard
                    key={idx}
                    {...bean}
                    userBeans={this.context.userBeans}
                  />
                  )
                })
              }
            </div>
        </div>
      </section>
    )
  }
}

