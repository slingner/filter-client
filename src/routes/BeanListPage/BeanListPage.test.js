import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BeanListPage from './BeanListPage'
import BeanListContext from '../../contexts/BeansListContext';


describe(`<BeanListPage />`, () => {
  //Snapshot Testing
  it('renders BeanListPage by default', () => {
    const wrapper = shallow(
    <BeanListContext.Provider>
      <BeanListPage />
    </BeanListContext.Provider>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})