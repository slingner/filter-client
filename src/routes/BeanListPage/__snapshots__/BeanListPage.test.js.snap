// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeanListPage /> renders BeanListPage by default 1`] = `<BeanListPage />`;
